### PR TITLE
Remove six dependency

### PR DESCRIPTION
**Pull request by RobbeSneyders**
This PR contains some clean-up work following the first phase of porting apache beam to Python 3.

- **Remove six dependency**
When starting the porting work, we decided to use the future package instead of the six package, since it allows us to port our code forward instead of back. This should simplify the future addition of Python 3 code and the refactoring if Python 2 support would ever be dropped.

- **Run py2to3 pylint on apache_beam module**
During porting, the list of modules to run this test on was extended with every ported module. Now that the phase has finished, we can run it on the whole apache_beam module.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by RobbeSneyders** [comment]
Run Python PreCommit

**Comment by RobbeSneyders** [comment]
R: @tvalentyn @Fematich

File: sdks/python/scripts/run_pylint_2to3.sh [hunk 1]
```diff
@@ -36,24 +36,16 @@ fi
 set -o errexit
 set -o pipefail
 
-DEFAULT_MODULE=apache_beam
+MODULE=$(ls -d -C apache_beam *.py)
 
-usage(){ echo "Usage: $0 [MODULE|--help]
-# The default MODULE is $DEFAULT_MODULE"; }
+usage(){ echo "Usage: $0 [MODULE|--help] # The default MODULE is MODULE"; }
```
Comment (tvalentyn) [inline sdks/python/scripts/run_pylint_2to3.sh]: Let's s/MODULE/$MODULE in the last occurrence on this line.

**Comment by RobbeSneyders** [comment]
I fixed the typo and removed the unused FUTURIZE_EXCLUDED.

**Comment by tvalentyn** [comment]
@charlesccychen could you merge this please?

**Comment by boyuanzz** [comment]
@charlesccychen This PR seems like ready to merge. Could you please help merge this?

**Comment by RobbeSneyders** [comment]
Run Python PreCommit

**Comment by charlesccychen** [comment]
Thanks! I will merge this after tests pass.  Can you also run the postcommit after the precommit passes?

**Comment by RobbeSneyders** [comment]
Run Python PostCommit

**Comment by RobbeSneyders** [comment]
Both tests passed. I just added a small change removing the six dependency from the dependency owners list.

**Comment by tvalentyn** [comment]
We can also remove `six` from https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile in this or follow-up PR.

**Comment by charlesccychen** [comment]
Thanks!

**Pull request merged**